src/instructlab/data/generate.py

```diff
@@ -198,6 +198,18 @@ def generate(
     from instructlab.sdg.generate_data import generate_data
     from instructlab.sdg.utils import GenerateException
 
+    # if --pipeline is not used, pipeline defaults to the value of ctx.obj.config.generate.pipeline
```

Comment:
I'm not sure this is a great direction. Are we going to add nulls to all other settings to "ensure --<option> needs to be used"? But I won't insist otherwise.

Reply:
This comment was left as a note to remember a conversation that Dan and I had about his previous comments on the PR. After we talked I understood why he wanted to make the default for `pipeline` and `Optional[str]` instead of a string and that was to retain the flexibility for the behavior described in this comment.

Is this behavior ideal, probably not, but we can determine what's best in the future.

Reply:
@booxter Do you have another recommendation on how to achieve the same result?  Or are you saying we must force a default to be specified?

Reply:
Thinking about this a bit more, is this any different than we handle backend for serving?  The only difference I can think of is we try to auto detect backend.  But if it's null in config, and autodetect doesn't work, you get an error.  And you need to specify --backend for force an option.